clippy_lints/src/literal_representation.rs

```diff
@@ -136,17 +155,22 @@ impl<'a> DigitInfo<'a> {
         };
 
         let mut last_d = '\0';
+        let len = sans_prefix.len();
         for (d_idx, d) in sans_prefix.char_indices() {
-            if !float && (d == 'i' || d == 'u') || float && (d == 'f' || d == 'e' || d == 'E') {
+            if !float && (d == 'i' || d == 'u') ||
+                float && (d == 'f' || d == 'e' || d == 'E') ||
+                (!float && is_possible_suffix_index(&d, d_idx, len)) {
                 let suffix_start = if last_d == '_' { d_idx - 1 } else { d_idx };
```

Comment:
Another interesting test case: `1__8`. The suggestion should be `1_i8`, but I think that it will be something different.

Reply:
I can probably add tests for `2__8` as well if we want to have lints for that as well. Don't think it will add too much complexity to this.

Reply:
> I think `is_possible_suffix_index` needs to check directly if the possible suffix is `"_8"`, `"_16"`, ... and not just if the length matches.

The check for those specific values are done using the `is_mistyped_suffix` function. So cases were the index function matches on e.g `_23` are discarded later on and the loop continues to check the next index.

Reply:
Yes please add a test for this. `2__8` gets currently linted (`consider: 28`). But should be `2_i8` in the future, while `2__9` should stay `consider: 29`

Reply:
> e.g _23 are discarded later on and the loop continues to check the next index.

Yes but I think that will fail on `__8`.

Reply:
It still returns `consider: 28` when you write `2__8`, so I think it does work as expected today. But that doesn't really matter now, since I'll be updating the handling of cases with double underscores and adding a few different test cases for those.

I'll add some extra tests for other strange uses of multiple underscores as well, just to document how we handle those.